tinyGS/src/Radio/Radio.cpp

```diff
@@ -35,16 +35,17 @@ Radio::Radio()
 
 void Radio::init()
 {
-  Log::console(PSTR("[SX12xx] Initializing ... "));
+Log::console(PSTR("[SX12xx] Initializing ... "));
   board_type board;
-  
+  bool useDefaultSPI = true;
+
   if (strlen(ConfigManager::getInstance().getBoardTemplate()) > 0)
   {
     size_t size = 512;
     DynamicJsonDocument doc(size);
     DeserializationError error = deserializeJson(doc, ConfigManager::getInstance().getBoardTemplate());
 
-    if (error.code() != DeserializationError::Ok || doc.containsKey("aADDR"))
```

Comment:
We need to do a better error handling here. If you only test for error code. The string " " will get a deserialization OK and will brick the board as it will continuously reboot and flashing again will not solve it as it is stored in the flash. (it can be fixed with a flash erasing tool but it's hard for some users)

The check for aDDR was a quick hack to avoid common mistakes. But we will have to check every parameter to be safe.

Reply:
Since the board config is well known could you not just have the inputs on the front end instead of inputting raw json string? 

At least that way you can sanitize the input before saving it?

Reply:
That was the first idea but it will make the config html way bigger (it is already quite big) and requires buffers for every parameter. And most importantly, if we need to add or remove a field we have to rebuild the config structure saved in flash which means resetting all boards config for every user or a complex updater to change the structure in flash before the config library loads.

The JSON makes it more flexible and also people can publish and share board templates and copy paste them easily for any board available on the market. The idea was taken from the tasmota project which does the same for every board:
https://templates.blakadder.com/ESP32_Lite_v1.html

A really good idea would be to have a separate html page to configure the template json string using a form. Again as tasmota does. All the validation could be done in js and the result sent in json to the board: https://tasmota.github.io/docs/Templates/

What do you think about that?

Reply:
Thats basically what i meant yea, create the json object on UI and POST it to esp to save it.